python/pylibcudf/pylibcudf/types.pyx

```diff
@@ -76,6 +76,9 @@ cdef class DataType:
     def __hash__(self):
         return hash((self.c_obj.id(), self.c_obj.scale()))
 
+    def __reduce__(self):
+        return (DataType, (self.c_obj.id(), self.c_obj.scale()))
```

Comment:
```suggestion
    def __reduce__(self):
        return (type(self), (self.c_obj.id(), self.c_obj.scale()))
```

Just in case these are ever subclassed.